src/briefcase/commands/new.py

```diff
@@ -327,6 +349,7 @@ def build_app_context(self):
 may only contain letters, numbers, hyphens and underscores; it can't contain
 spaces or punctuation, and it can't start with a hyphen or underscore.
 
+Due to build requirements, the app name may also not be a Java or Python keyword.
```

Comment:
The guidance here should say to "app name should avoid reserved words in common programming languages".